includes/class-sensei-guest-user.php

```diff
@@ -137,7 +146,8 @@ public function init() {
 		add_action( 'sensei_is_enrolled', [ $this, 'open_course_always_enrolled' ], 10, 3 );
 		add_action( 'sensei_can_access_course_content', [ $this, 'open_course_enable_course_access' ], 10, 2 );
 		add_action( 'sensei_can_user_manually_enrol', [ $this, 'open_course_user_can_manualy_enroll' ], 10, 2 );
-		add_action( 'sensei_send_emails', [ $this, 'skip_sensei_email' ] );
+		add_filter( 'sensei_send_emails', [ $this, 'skip_sensei_email' ] );
+		add_filter( 'pre_wp_mail', [ $this, 'skip_wp_mail' ], 10, 2 );
```

Comment:
Notice that I decided to add a `pre_wp_mail` filter for guest users and one for preview users.

The idea is that the user might want - for some reason - to disable one or another. I'm not sure why someone would want to do that. But maybe someone can consider that preview users should receive e-mails as usual, while guest users shouldn't? Not sure.

I can quickly change that to a single `pre_wp_mail` filter, but, given that most of the logic is shared, for me it seems OK to leave as it is. :)